src/media/track.ts

```diff
@@ -0,0 +1,220 @@
+/* eslint-disable no-underscore-dangle */
+import { EventMap } from 'typed-emitter';
+import EventEmitter from '../event-emitter';
+import { logger } from '../util/logger';
+
+export enum Events {
+  Ended = 'ended',
+  Muted = 'muted',
+  /**
+   * Fires when the published state of a LocalTrack changes.  A value of 'true' indicates the
+   * track has been published, and 'false' indicates that it is no longer published.  Tracks are
+   * unpublished by default, and must be published explicitly (no event will be fired to indicate
+   * the initial state of 'unpublished').
+   */
+  PublishedStateUpdate = 'published-state-update',
+  /**
+   * Fires when there has been a change in the underlying track.
+   */
+  UnderlyingTrackChange = 'underlying-track-change',
+}
+
+export interface TrackState {
+  id: string;
+  label: string;
+  muted: boolean;
+}
+
+export type TrackEndEvent = {
+  trackState: TrackState;
+};
+
+export type TrackMuteEvent = {
+  trackState: TrackState;
+};
+// eslint-disable-next-line no-shadow
+export enum TrackStatus {
+  ENDED = 'ended',
+  LIVE = 'live',
+}
+
+// eslint-disable-next-line no-shadow
+export enum TrackKind {
+  AUDIO = 'audio',
+  VIDEO = 'video',
+}
+
+export type TrackPublishEvent = {
+  isPublished: boolean;
+  trackState: TrackState;
+};
+
+export interface TrackEvents extends EventMap {
+  [Events.Ended]: (event: TrackEndEvent) => void;
+  [Events.Muted]: (event: TrackMuteEvent) => void;
+  [Events.PublishedStateUpdate]: (event: TrackPublishEvent) => void;
+  [Events.UnderlyingTrackChange]: () => void;
+}
+
+export type TrackEffect = any;
+// TBD: Fix this once types are published seperatly
+// export type TrackEffect = BaseMicrophoneEffect | BaseCameraEffect;
+
+/**
+ * Basic Track class. Wrapper for LocalTrack from 'webrtc-core'.
+ */
+export abstract class Track extends EventEmitter<TrackEvents> {
+  static Events = Events;
+
+  isPlaying: boolean;
```

Comment:
Is this needed? The track state is already modeled by the underlying track, no?

Reply:
I see now this refers to the element, see my comment on that below.

Reply:
This to if someone does a play and pass src element we keep the status if its playing actively

Reply:
this can be removed now, right?